liam2/tests/test_liam2.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+
+
+from __future__ import print_function
+
+import os
+import pkg_resources
+
+import liam2
+from liam2.simulation import Simulation
+
+
+def run_simulation(yaml_file, input_dir, output_dir):
+    try:
+        simulation = Simulation.from_yaml(
+            yaml_file,
+            input_dir = input_dir,
+            output_dir = output_dir,
+            )
+        simulation.run(False)
+    except Exception:
+        print('{} failed'.format(yaml_file))
+        raise
+
+
+def test_liam2_examples_files():
+    liam2_examples_directory = os.path.join(
+        pkg_resources.get_distribution('liam2').location,
+        'liam2',
+        'tests',
+        'examples'
+        )
+    excluded_files = [
+        'demo_import.yml',  # import file
+        'demo02.yml',  # TODO: pb with figures
+        ]
+    yaml_files = [os.path.join(liam2_examples_directory, _file) for _file in os.listdir(liam2_examples_directory)
+        if os.path.isfile(os.path.join(liam2_examples_directory, _file))
+        and _file.endswith('.yml')
+        and _file not in excluded_files]
+
+    input_dir = os.path.join(liam2_examples_directory)
+    output_dir = os.path.join(
+        pkg_resources.get_distribution('liam2').location,
+        'liam2',
+        'tests',
+        'output',
+        )
+    for yaml_file in yaml_files:
+        yield run_simulation, yaml_file, input_dir, output_dir
+
+
+def test_liam2_functionnal_files():
+    liam2_functional_directory = os.path.join(
+        pkg_resources.get_distribution('liam2').location,
+        'liam2',
+        'tests',
+        'functional'
+        )
+    # TODO does not work
+    import_yaml_path = os.path.join(liam2_functional_directory, 'import.yml')
+    assert os.path.exists(import_yaml_path), '{} does not exists'.format(import_yaml_path)
+    simulation_yaml_path = os.path.join(liam2_functional_directory, 'simulation.yml')
+    assert os.path.exists(simulation_yaml_path), '{} does not exists'.format(simulation_yaml_path)
+
+    input_dir = os.path.join(liam2_functional_directory)
+    output_dir = os.path.join(
+        pkg_resources.get_distribution('liam2').location,
+        'tests',
+        'output'
+        )
+    yaml_files = [import_yaml_path, simulation_yaml_path]

```

Comment:
Hi everybody,

I worked on the test_liam2 file. It's lost somewhere in a field of fork and
pull request:
you can see it there :

https://github.com/AlexisEidelman/liam2/blob/convergence/liam2/tests/test_liam2.py

It enables to run 'import' yaml files.



2015-08-25 15:20 GMT+02:00 Gaëtan de Menten <notifications@github.com>:

> In liam2/tests/test_liam2.py
> <https://github.com/liam2/liam2/pull/182#discussion_r37863283>:
>
> > +        'tests',
> > +        'functional'
> > +        )
> > +    # TODO does not work
> > +    import_yaml_path = os.path.join(liam2_functional_directory, 'import.yml')
> > +    assert os.path.exists(import_yaml_path), '{} does not exists'.format(import_yaml_path)
> > +    simulation_yaml_path = os.path.join(liam2_functional_directory, 'simulation.yml')
> > +    assert os.path.exists(simulation_yaml_path), '{} does not exists'.format(simulation_yaml_path)
> > +
> > +    input_dir = os.path.join(liam2_functional_directory)
> > +    output_dir = os.path.join(
> > +        pkg_resources.get_distribution('liam2').location,
> > +        'tests',
> > +        'output'
> > +        )
> > +    yaml_files = [import_yaml_path, simulation_yaml_path]
>
> not enough. There are other models in functional that needs testing
> (variant, generate, ...)
> See https://github.com/liam2/liam2/blob/master/make_release.py#L297
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/liam2/liam2/pull/182/files#r37863283>.
>